post-scan-actions/aws-python-slack-notification/handler.py

```diff
@@ -0,0 +1,45 @@
+import urllib3
+import json
+import os
+http = urllib3.PoolManager()
+from botocore.exceptions import ClientError
+
+def lambda_handler(event, context):
+
+    url = os.environ['SLACK_URL']
+    channel = os.environ['SLACK_CHANNEL']
+    username = os.environ['SLACK_USERNAME']
+    
+    for record in event['Records']:
+
+        #Message details from SNS event
+        message = json.loads(record['Sns']['Message'])
+        findings = message['scanning_result'].get('Findings')
+
+        #ARN info to get AWS Account ID
+        arn = json.dumps(record['EventSubscriptionArn'])
+        account_id = arn.split(":")[4].strip()
+
+        if findings:
+        
+            for finding in message['scanning_result']['Findings']:
```

Comment:
Although you loop the findings array, but in the end only the last finding exists in the message. Try collecting all the findings before formatting the message.

Reply:
Email integration example appends to the body_text in each iteration of the for loop. This loop replaces the body_text in each iteration so only the last finding makes it to the payload.

Reply:
I fixed doing this way here:
if findings:
        
            malware = json.dumps(message['scanning_result']['Findings'][0]['malware'])
            type_malware =json.dumps(message['scanning_result']['Findings'][0]['type'])

            body_text = textwrap.dedent('''
            WARNING
            AWS Account ID: {account}
            File URL: {file_url}
            Malware Name: {malware}
            Malware Type: {type}
            ''').format(account=str(account_id),
                    file_url=str(message['file_url']),
                    malware=str(malware),
                    type=str(type_malware))
            
Any recommendation?

Reply:
But then you only handle the first (0) finding. The Findings is any array in case that there are multiple malware detections in a file, say a zip. You should be looping through the findings and concat the malware names.
Some pseudo code for you to reference:
```python
malwares = ''
types = ''
for finding in message['scanning_result']['Findings']:
    malwares += finding.get('malware')
    types += finding.get('type')

body_text = textwrap.dedent('''
WARNING
AWS Account ID: {account}
File URL: {file_url}
Malwares Name: {malwares}
Malwarse Type: {types}
''').format(account=str(account_id),
        file_url=str(message['file_url']),
        malwares=malwares,
        types=types)
```

Reply:
Correct. Because it's being based on every SNS notification. My fear is doing the way how you share grab some duplicated info in the queue. But I could be wrong.
If you think this will be the best way to do it, I'm ok  with it

Reply:
I'm not sure what you mean by `in the queue` but I am positive that it wouldn't have duplicated info.

Reply:
Ok. Sounds good to me

Reply:
@carpusherw This way that you recommended didn't work:

malwares = ''
types = ''
for finding in message['scanning_result']['Findings']:
    malwares += finding.get('malware')
    types += finding.get('type')

body_text = textwrap.dedent('''
WARNING
AWS Account ID: {account}
File URL: {file_url}
Malwares Name: {malwares}
Malwarse Type: {types}
''').format(account=str(account_id),
        file_url=str(message['file_url']),
        malwares=malwares,
        types=types)

------------------------------------------------------------------------------------------
Can I keep this way here that is working?

malware = json.dumps(message['scanning_result']['Findings'][0]['malware'])
type_malware =json.dumps(message['scanning_result']['Findings'][0]['type'])

body_text = textwrap.dedent('''
WARNING
AWS Account ID: {account}
File URL: {file_url}
Malware Name: {malware}
Malware Type: {type}
''').format(account=str(account_id),
        file_url=str(message['file_url']),
        malware=str(malware),
        type=str(type_malware))

------------------------------------------------------------------------------------------
Or do you have any additional recommendations?

Reply:
That was just _pseudo code_ as I mentioned. But no problem, here is a working version that I've tested.
```python
malwares = []
types = []
for finding in message['scanning_result']['Findings']:
    malwares.append(finding.get('malware'))
    types.append(finding.get('type'))

body_text = textwrap.dedent('''
    WARNING
    AWS Account ID: {account}
    File URL: {file_url}
    Malware Name(s): {malwares}
    Malware Type(s): {types}
''').format(
    account=str(account_id),
    file_url=str(message['file_url']),
    malwares=', '.join(malwares),
    types=', '.join(types)
)
```

Reply:
Awesome. Thank you!!!